### PR TITLE
Deployment

### LINKED ISSUE #11: Deploy BE with Heroku
---
Name: Deploy with Heroku
About: host api publicly

---

### User Story
chore

### PR DESCRIPTION
**Pull request by dtsehl**
# Description
* Get deployment working on Heroku - https://careerdaybe.herokuapp.com/
  * Endpoints are as follows: 
      * /api/v1/jobs - gets a list of all jobs 
      * /api/v1/jobs/<job_id> - get a larger set of information about one particular job
  * Add a `config.py` file which tells the application, depending on which environment is being run, the value of various environment variables

# Closes issue(s)
#11 

# How to test / reproduce
Follow what is already outlined in the README - the endpoints above should work on Heroku.

# Screenshots
Screen Shot 2020-10-26 at 3 16 21 PM

Screen Shot 2020-10-26 at 3 18 32 PM

# Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist
- [ ] I have tested this code
- [ ] I have updated the Readme

# Other comments

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-app.config['SQLALCHEMY_DATABASE_URI'] = "postgresql://postgres:postgres@localhost:5432/career_day"
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('SQLALCHEMY_DATABASE_URI')
 db = SQLAlchemy(app)
 migrate = Migrate(app, db)
 
-if __name__ == "__main__":
-    app.run(debug=True)
+if __name__ == '__main__':
+    # port = int(os.environ.get('PORT', 5000))
+    app.run()
```
Comment (Strompy) [inline app.py:13]: I'm glad we got this worked out

File: app.py [hunk 2]
```diff
@@ -243,7 +245,7 @@ def handle_jobs():
 
     return {"jobs": results}
 
-@app.route('/job/<job_id>')
+@app.route('api/v1/jobs/<job_id>')
```
Comment (janegreene) [inline app.py]: Updated for versioning capabilities.

Comment (Strompy) [inline app.py]: This is a good update in case of future versioning

**Review by Strompy — APPROVED** [review]

**Review by janegreene — APPROVED** [review]
Heroku and local deployment

File: app.py (hunk 2 above)
Comment (janegreene) [reply app.py]: Plus 1,  good update for consistency 'job' vs 'jobs'.

**Review by Strompy — APPROVED** [review]

File: app.py [hunk 3]
```diff
@@ -1,14 +1,16 @@
 from flask import Flask, request
 from flask_sqlalchemy import SQLAlchemy
 from flask_migrate import Migrate
+import os
 
 app = Flask(__name__)
-app.config['SQLALCHEMY_DATABASE_URI'] = "postgresql://postgres:postgres@localhost:5432/career_day"
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('SQLALCHEMY_DATABASE_URI')
```
Comment (janegreene) [inline app.py:7]: Heroku has remote postgresQL connected in config vars.

**Review by janegreene — APPROVED** [review]
Versioned routes and deployable to heroku and local development.

**Pull request merged**